### ISSUE TITLE
Running npm ci without deleting old node_modules folder

### ISSUE THREAD

**Issue by michaeluskov** [npm/npm#20104]
#### What's the feature?

`npm ci` is great to use on CI server. But now deleting `node_modules` folder + running npm ci is much slower on Windows machine than just running `npm install `.

For instance, `npm install`:

> up to date in 8.125s

`npm ci`:

> added 533 packages from 702 contributors in 33.828s

Is there an opportunity to add some argument (`--no-delete` for example) to `npm ci` command?

#### What problem is the feature intended to solve?

Faster CI builds.

#### Is the absence of this feature blocking you or your team? If so, how?

This runs way slower so we're blocked from using it. 

#### Is this feature similar to an existing feature in another tool?

Yes, this is implemented in our local .NET package manager.

#### Is this a feature you're prepared to implement, with support from the npm CLI team?

Yes.

**Comment by bosonojka5** [comment]
I have this issue too.

**Comment by simonua** [comment]
I like the intent of this suggestion, @michaeluskov. I have a similar need and used `npm ci` locally to speed up installs, but I haven't been as happy taking that approach as I see `npm ci` really just for CI environments.

I wonder if the approach should be different as I see this suggestion change the purpose of `npm ci`, which, I believe, is solely or at least primarily intended towards an, ideally, clean CI build with no preexisting _node_modules_ folder. In short, I believe, it should do one standard task very well and very consistently. 

Rather than modifying the identity of this new command, perhaps it would be better to enhance `npm i|install` to accept a flag to install strictly from _package-lock.json_ and a secondary flag to ignore _node_modules_. I think that would achieve the same result but from a different angle. If popular, perhaps another command alias could be added.

**Comment by davideberlein** [comment]
I agree, a --no-delete would really help to speed up situations where the installed node_modules already are in sync with the package-lock.json.

In our case a webpack server is started in the background when our Java Jetty server is started in Eclipse. When the project is initially checked out, we need to do a npm ci. As long as no one changes the packe-lock.json we wouldn't need to delete-and re-install everything on each start of our Jetty Server. a `--no-delete` or `--no-clean` or `--clean=false` option would be very helpful here.

**Comment by RobCannon** [comment]
I have found another scenario where having a --no-delete option would be helpful.

I am developing locally with docker and my source directory is mapped as a volume into the docker file.  In this case, I want docker to always have a clean directory and not my local files so I can be sure it will match the results of our CI build.  So, I have a volumes mapped like this:
```
    volumes:
      - ./client:/root/client
      - /root/client/node_modules
```
Docker will mount an empty directory into node_modules, regardless of what is present on my local drive.  BUT, npm ci will try to delete that directory (even though it is empty), but it can't because docker has that directory locked.  npm then fails.
